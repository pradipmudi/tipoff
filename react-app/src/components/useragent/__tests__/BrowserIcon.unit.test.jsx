import React from "react";
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import TestBrowserIcon from '../BrowserIcon';

describe("BrowserIcon", () => {

    it("matches snapshot", () => {
        const browser={
            name : 'Chrome',
            version : '79.12',
            major : '79'
        }
        const tree = renderer.create(<TestBrowserIcon browser={browser} />).toJSON();
        expect(tree).toMatchSnapshot();
    
    });
    
    it("renders without issues", () =>{
        const browser={
            name : 'Chrome',
            version : '79.12',
            major : '79'
        }
        const component = shallow(<TestBrowserIcon browser={browser} />);
        expect(component.find('.fab').hasClass('fa-chrome')).toEqual(true);
        expect(component.find('.fab').props().title).toEqual(`${browser.name}:${browser.major}`);
    });

    it("renders with default icon when browser is not passed", () => {
        const component = shallow(<TestBrowserIcon  />);
        expect(component.find('.far').hasClass('fa-window-maximize')).toEqual(true);
    })

});

