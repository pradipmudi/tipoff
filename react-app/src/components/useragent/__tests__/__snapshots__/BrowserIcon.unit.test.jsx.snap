// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowserIcon matches snapshot 1`] = `
<span>
  <span
    className="icon"
    title="Chrome:79"
  >
    <i
      aria-hidden="true"
      className="fab fa-chrome"
    />
  </span>
</span>
`;
